misk/src/main/kotlin/misk/concurrent/ExplicitReleaseBlockingQueue.kt

```diff
@@ -0,0 +1,80 @@
+package misk.concurrent
+
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.Delayed
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.PriorityBlockingQueue
+import java.util.concurrent.TimeUnit
+
+/**
+ * An [ExplicitReleaseBlockingQueue] is a [BlockingQueue] that only returns elements from
+ * [BlockingQueue.take], [BlockingQueue.poll], and [BlockingQueue.peek] when explicitly
+ * released by the application. Useful for tests that want to explicitly control when
```

Comment:
Just curious, is this only meant for tests? Should we be more clear about that?

Reply:
Sure - is there wording you'd want to see beyond the comment above?